int32 data1
string data2